python/lsst/daf/butler/_butler.py

```diff
@@ -1736,6 +1736,39 @@ def _clone(
         """
         raise NotImplementedError()
 
+    @abstractmethod
+    def redefine_collection_chain(
+        self, parent_collection_name: str, child_collection_names: str | Iterable[str]
+    ) -> None:
+        """Replace the contents of a CHAINED collection with new children.
+
+        Parameters
+        ----------
+        parent_collection_name : `str`
+            The name of a CHAINED collection to which we will assign new
+            children.
+        child_collection_names : `Iterable` [ `str ` ] | `str`
```

Comment:
```suggestion
        child_collection_names : `~collections.abc.Iterable` [ `str ` ] | `str`
```
otherwise Sphinx doesn't recognize it, unfortunately.